controllers/application_controller.go

```diff
@@ -196,11 +209,24 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{}, err
 		}
 
+		updateRequired := false
+		// nil out the attributes and projects for the application devfile
+		// The Attributes contain any image components for the application
+		// And the projects contains any git components for the application
+		devWorkspacesSpec := devfileData.GetDevfileWorkspaceSpec().DeepCopy()
+		devWorkspacesSpec.Attributes = nil
+		devWorkspacesSpec.Projects = nil
```

Comment:
wait why do we do this again

Reply:
To make comparisons of the added git projects (under projects in the devfile spec) and container image projects (under attributes) easier in the devfile app model. Basically we create a copy of the devworkspace spec, nil out the attributes and projects, add the components to it, and compare with the existing devworkspace spec.